src/main/java/frc/robot/lib/ControllerDrive.kt

```diff
@@ -0,0 +1,92 @@
+package frc.robot.lib
+
+import edu.wpi.first.math.MathUtil
+import edu.wpi.first.units.measure.AngularVelocity
+import edu.wpi.first.units.measure.LinearVelocity
+import edu.wpi.first.wpilibj.XboxController
+import frc.robot.lib.bindings.DRIVE_MODIFIERS
+import frc.robot.subsystems.Intake
+import frc.robot.subsystems.drivetrain.Chassis
+import org.littletonrobotics.junction.Logger
+
+private const val UP_ADJUST = 0.5
+private const val DOWN_ADJUST = 0.4
+private const val JOYSTICK_DEADBAND = 0.1
+private const val TRIGGER_DEADBAND = 0.05
+private val maxTranslation = 3.metersPerSecond
+private val maxRotation = 1.rotationsPerSecond
+
+/* Extension properties to access the drive velocities based on joystick input
+ * This used to be a single method that returned a ChassisSpeeds object,
+ * but it made it hard to use in cases where we wanted an override of one or more speeds,
+ * such as auto-alignment */
+
+/**
+ * Calculates a speed modifier that allows scaling up or down based on the values of the left and
+ * right triggers. The output is between (1-UP_ADJUST-DOWN_ADJUST) and 1
+ */
+val XboxController.triggerAdjust
+    get() =
+        (1 - UP_ADJUST) + (MathUtil.applyDeadband(rightTriggerAxis, TRIGGER_DEADBAND) * UP_ADJUST) -
+            (MathUtil.applyDeadband(leftTriggerAxis, TRIGGER_DEADBAND) * DOWN_ADJUST)
+
+val obstacleSlowdown
+    get() =
+        distanceSlowdown(
+                FieldGeometry.distanceToClosestCoralStation(Chassis.state.Pose.translation).also {
+                    Logger.recordOutput("closestCoralStationDistance", it)
+                },
+                1.5,
+                1.0,
+                0.5,
```

Comment:
kotlin lets you add names to arguments. 
```
a = 1.5
b = 1.0
c = 0.5
```

We should add them in here to improve legibility